simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/BufferPoolsExports.java

```diff
@@ -0,0 +1,128 @@
+package io.prometheus.client.hotspot;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.GaugeMetricFamily;
+
+import java.lang.management.ManagementFactory;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * A backward compatible way to get the buffer MX beans as well
+ *
+ * Can be replaced with a simple access once JDK 1.7 compatibility is baseline.
+ *
+ * @author marcus
+ * @since 1.0.0
+ */
+public class BufferPoolsExports extends Collector {
+
+    private static final Logger LOGGER = Logger.getLogger(BufferPoolsExports.class.getName());
+
+    private final List<Object> bufferPoolMXBeans = new ArrayList<Object>();
+    private Method getName;
+    private Method getMemoryUsed;
+    private Method getTotalCapacity;
+    private Method getCount;
+
+    public BufferPoolsExports() {
+        try {
+            final Class<?> bufferPoolMXBeanClass = Class.forName("java.lang.management.BufferPoolMXBean");
+            bufferPoolMXBeans.addAll(accessBufferPoolMXBeans(bufferPoolMXBeanClass));
+
+            getName = bufferPoolMXBeanClass.getMethod("getName");
+            getMemoryUsed = bufferPoolMXBeanClass.getMethod("getMemoryUsed");
+            getTotalCapacity = bufferPoolMXBeanClass.getMethod("getTotalCapacity");
+            getCount = bufferPoolMXBeanClass.getMethod("getCount");
+
+        } catch (ClassNotFoundException e) {
+            LOGGER.info("BufferPoolMXBean not available, not metrics for buffer pools will be exported");
```

Comment:
I'm not sure we should be putting this in the logs, just debug is fine.

s/not/no/